OpenSim/Simulation/Model/ModelComponent.h

```diff
@@ -108,6 +109,38 @@ OpenSim_DECLARE_ABSTRACT_OBJECT(ModelComponent, Component);
      */
     Model& updModel();
 
+    /** Perform any computations that must occur before the Model is scaled. For
+        example, a GeometryPath must calculate and store its path length in the
+        current body position before scaling so that an owning Muscle can use
+        this information to update the properties of the muscle after scaling.
+        This method is virtual and may be implemented by any subclass of
+        ModelComponent, but all implementations must begin with a call to
+        `Super::preScale()` to execute the parent class methods before the child
+        class method. The base class implementation in ModelComponent does
+        nothing.
+        @see scale()
+        @see postScale() */
+    virtual void preScale(const SimTK::State& s, const ScaleSet& scaleSet) {};
+
+    /** Scale the ModelComponent. This method is virtual and may be implemented
+        by any subclass of ModelComponent, but all implementations must begin
+        with a call to `Super::scale()` to execute the parent class methods
+        before the child class method. The base class implementation in
+        ModelComponent does nothing.
+        @see preScale()
+        @see postScale() */
+    virtual void scale(const SimTK::State& s, const ScaleSet& scaleSet) {};
+
+    /** Perform any computations that must occur after the Model has been
+        scaled. This method is virtual and may be implemented by any subclass of
```

Comment:
"after the Model has been scaled" is a little vague to me, since "postScale" really is still scaling the model. Consider rephrasing as "after ModelComponent::scale() has been invoked on all components of the model."

Reply:
Done. Also updated the analogous comment for `ModelComponent::preScale()`, and changed the comment "Scale the model." to "Scale the rest of the model." in `Model::scale()`.